SignupProcessingScreen refinements (Fix to #6114)

coreh opened the pull request:
This PR fixes Issue #6114, by making it so that the `SignupFlowController` is reset before the user clicks on the Continue button.

It also adds two other improvements:

- The `SignupProcessingScreenI18n` stub is removed (We no longer need that, since the actual component landed)
- The `Sign Up` link in the masterbar is no longer displayed during signup. (So that user's can't navigate to `/signup` by accidentally clicking on it)

Test live: https://calypso.live/?branch=fix/issue-6114

michaeldcain commented on client/layout/masterbar/logged-out.jsx:
```diff
@@ -13,7 +13,7 @@ import { localize } from 'i18n-calypso';
 import config from 'config';
 
-const MasterbarLoggedOut = ( { title, translate } ) => (
+const MasterbarLoggedOut = ( { title, translate, showSignup = true } ) => (
```
Let's add `showSignup` to `.propTypes` in L44, and although I know they achieve the same thing, we should probably either define the default in `defaultProps` (L48 - like `title`), and remove it here, or define `title = ''` here, and remove `.defaultProps` for consistency.

coreh replied:
Oh I missed that entirely. Adding it.

michaeldcain commented:
@coreh: Just a small edit, but otherwise is looks and works great. Thanks for the quick fix!

coreh commented:
Fixed. I also moved `translate` to the end of the props object destructuring, just for organization's sake, since it seems to be added by the `localize` function, at the bottom.

michaeldcain commented:
LGTM
